src/components/ec/cuda/kernel/ec_cuda_reduce.cu

```diff
@@ -0,0 +1,336 @@
+                }                                                               \
+            }                                                                   \
+            break;                                                              \
+        }                                                                       \
+        if (flags & UCC_EEE_TASK_FLAG_REDUCE_WITH_ALPHA) {                      \
+            for (i = start; i < count; i += step) {                             \
+                d[i] = d[i] * (_AlphaType)task.alpha;                           \
+            }                                                                   \
+        }                                                                       \
+    }
+
+#define CUDA_REDUCE_WITH_OP_STRIDED(NAME, _OP)                                 \
+    template <typename _Type, typename _AlphaType>                             \
+    __global__ void UCC_REDUCE_CUDA_STRIDED_##NAME(                            \
+        const _Type *s1, const _Type *s2, _Type *d, size_t count,              \
+        size_t stride, uint16_t n_src2, const bool with_alpha,                 \
+        const float alpha)                                                     \
+    {                                                                          \
+        size_t start = blockIdx.x * blockDim.x + threadIdx.x;                  \
+        size_t step  = blockDim.x * gridDim.x;                                 \
+        size_t ld    = stride / sizeof(_Type);                                 \
+        size_t i;                                                              \
+                                                                               \
+        ucc_assert(stride % sizeof(_Type) == 0);                               \
+        switch (n_src2) {                                                      \
+        case 1:                                                                \
+            for (i = start; i < count; i += step) {                            \
+                d[i] = _OP##_2(s1[i], s2[i]);                                  \
+            }                                                                  \
+            break;                                                             \
+        case 2:                                                                \
+            for (i = start; i < count; i += step) {                            \
+                d[i] = _OP##_3(s1[i], s2[i], s2[i + ld]);                      \
+            }                                                                  \
+            break;                                                             \
+        case 3:                                                                \
+            for (i = start; i < count; i += step) {                            \
+                d[i] = _OP##_4(s1[i], s2[i], s2[i + ld], s2[i + 2 * ld]);      \
```

Comment:
Would it be preferable to move the striding logic to ucc_dt_reduce_strided in ucc_dt_reduce.h (as proposed in a previous comment)? Maybe we would avoid duplicating CUDA_REDUCE_WITH_OP for the strided version

Reply:
not sure what do you mean by "move striding logic to ucc_dt_reduce_striedd" ? DO you mean converting to default reduce ? Then, again, the answer is that we want to keep task_reduce_strided for better signalling in ee_cuda.

Reply:
yes, this is what I meant. Ok for the explanation